Lib/multiprocessing/synchronize.py

```diff
@@ -270,13 +270,13 @@ def wait(self, timeout=None):
 
     def notify(self, n=1):
         assert self._lock._semlock._is_mine(), 'lock is not owned'
-        assert not self._wait_semaphore.acquire(False)
+        assert not self._wait_semaphore.acquire(False), 'Nobody waiting?'
```

Comment:
Perhaps this is an overlook, but the message here is still wrong (it's ok to have nobody waiting).

Reply:
Hrm. So exactly what is the assertion testing for? Thanks for the clarification!

Reply:
It allows waking up the exact desired number of waiters (see below). The logic is a bit difficult to understand, as the various semaphores don't have an obvious meaning, they are just used for careful synchronization between notifier and waiters.

Reply:
"notify: Should not have succeeded at acquiring _wait_semaphore"?

Reply:
Sounds ok.

Reply:
Done (slightly shorter version).